### PR TITLE
Allow updates of secrets from 1Password

### PR DESCRIPTION
**Pull request by spenczar**
This changes the installer/generate_secrets.py script to allow updates of secrets that already exist in Vault, and have a value in 1Password. Previously, the value in Vault was always preserved.

There is some special behavior for ingress-nginx and cert-manager which I don't understand; I'm preserving the old code's behavior (never update those two) just out of caution.

### CODE REVIEW

**Comment by spenczar** [comment]
I noticed that for data-int.lsst.cloud, this will expand the `butler` vault secret to have a lot more inside it, since 1Password has many changes that have not been reflected into Vault yet. I think @rra is the last one to have touched the butler secrets in RSP-Vault, if I'm reading the 1Password activity history stuff correctly. Russ, can you confirm whether syncing Vault to the current contents of 1Password is alright?

**Review by jonathansick — APPROVED** [review]
Looks good. I can see this working (although I do struggle to understand these scripts sometimes).

File: installer/generate_secrets.py [hunk 1]
```diff
@@ -382,31 +382,23 @@ def generate(self):
...
+        for composite_key, secret_value in self.op_secrets.items():
+            item_component, item_name = composite_key.split()
+            # Special case for components that may not be present in every
+            # environment, but nonetheless might be 1Password secrets (see
+            # conditional in SecretGenerator.generate)
+            if item_component in {"ingress-nginx", "cert-manager"}:
+                continue
+
+            logging.debug("Updating component: %s/%s", item_component, item_name)
+            self.secrets[item_component][item_name] = secret_value
```
[the remark this answers is not in the document]
Comment (spenczar) [reply installer/generate_secrets.py:401]: I guess so. I felt that using `input_field` for this felt a little opaque, like it kind of hides what's going on, but it _is_ what all the rest of the code does, so I should probably follow the pattern that is used everywhere else.

**Comment by rra** [comment]
I'm not sure what changes to the butler secret you're talking about, but my guess is that they're related to adding the necessary information for vo-cutouts to function. But that was working fine in data-int, so those changes were definitely already made in Vault.

Without more specifics, I'm afraid I don't know what you're seeing.

**Comment by spenczar** [comment]
I'm hesitant to post the specifics here since it involves secrets; I'll send you a DM over slack @rra.

**Pull request merged**